### PR TITLE
Admin Contract Signing

### PR DESCRIPTION
**Pull request by tanatorn**
* Creates an endpoint - `admin#partnered_signup_sign_document` - to handle redirect to docusign from admin for signing of the contract
* Adds a role to the docusign redirect endpoint to actually support both the signer role and the administrator role
Here's the full flow:

There's some state management left to do, e.g. we need to move the date of `submitted_at`, `accepted_at` and `rejected_at` to a state machine model, but I figured we can do that in another PR

### CODE REVIEW

File: app/services/partners/docusign/admin_contract_signing.rb
Comment (tanatorn) [inline app/services/partners/docusign/admin_contract_signing.rb]: @maxwofford  I need the email/name you put for the `admin` role for the docusign template.I'm currently getting this error:
```
{"errorCode":"UNKNOWN_ENVELOPE_RECIPIENT","message":"The recipient you have identified is not a valid recipient of the specified envelope."}
```

File: app/controllers/partnered_signups_controller.rb
Comment (tanatorn) [inline app/controllers/partnered_signups_controller.rb]: this seems to be a bug with the partnered signups cc: @eilla1

Comment (exu3) [reply app/controllers/partnered_signups_controller.rb]: The partnered signup address line (previously just `owner_address`) was separated into 6 fields in #2061 ([migration](https://github.com/hackclub/bank/blob/main/db/migrate/20211231074013_add_address_lines_to_partnered_signups.rb)). Did that cause an issue? I can't seem to tell 😅 It's possible I could have done something wrong or missed a step.

Comment (tanatorn) [reply app/controllers/partnered_signups_controller.rb]: I wasn't able to create a new partnered signup as it was looking for `owner_address` that wasn't there, we needed to add this change and the one in the partnered_signups model

Comment (exu3) [reply app/controllers/partnered_signups_controller.rb]: Ahhh sorry about that. I made a PR #2199 to fix it so that the form works again for anyone who needs to work with it.

File: app/services/partners/docusign/admin_contract_signing.rb
Comment (maxwofford) [reply app/services/partners/docusign/admin_contract_signing.rb]: The template doesn't have anything in the name or email field. Maybe this is in test mode and you can only send to the developer account? The email would be max@hackclub.com in that case.

Screen Shot 2022-01-09 at 21 51 33

**Comment by tanatorn** [comment]
PTAL @maxwofford @garyhtou

**Comment by maxwofford** [comment]
This looks great! If this interferes with the partnered signup flow (sending "connect" to docusign, then to an error flash) we can't merge it just yet while givebutter is still testing the flow on their end. Could we have an exception for partner id#2 before merge?

**Review by maxwofford — APPROVED** [review]
https://github.com/hackclub/bank/pull/2198#issuecomment-1012177334 Go ahead and merge once this is hidden for partner id#2

**Comment by tanatorn** [comment]
@maxwofford this shouldn't block partner 2 since we have this conditional logic here - https://github.com/hackclub/bank/blob/main/app/controllers/partnered_signups_controller.rb#L26

**Pull request merged**
